keanu-python/keanu/algorithm/sampling.py

```diff
@@ -20,3 +20,16 @@ def sample(net, sample_from, algo='metropolis', draws=500, drop=0, down_sample_i
     vertex_samples = {Vertex._get_python_id(vertex_unwrapped): list(map(Tensor._to_ndarray, network_samples.get(vertex_unwrapped).asList())) for vertex_unwrapped in vertices_unwrapped}
 
     return vertex_samples
+
+def iterate_samples(net, sample_from, algo='metropolis', drop=0, down_sample_interval=1):
+	vertices_unwrapped = k.to_java_object_list(sample_from)
+
+	sample_iterator = algorithms[algo].withDefaultConfig().generatePosteriorSamples(net.unwrap(), vertices_unwrapped).dropCount(drop).downSampleInterval(down_sample_interval).stream().iterator()
+	
+	return _iterator_samples(sample_iterator, vertices_unwrapped)
+
+def _iterator_samples(sample_iterator, vertices_unwrapped):
+	while (True):
```

Comment:
What happens when you reach the end of the iterator? Don't you need to loop `while (sample_iterator.hasNext())`?

Reply:
There's no end to the iterator - that's the beauty of it! It will keep sampling for as long as the user wants.